src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -371,5 +406,12 @@ private static int GetTokenTypeIndex(string classificationType, Dictionary<strin
             Contract.ThrowIfFalse(tokenTypesToIndex.TryGetValue(tokenTypeStr, out var tokenTypeIndex), "No matching token type index found.");
             return tokenTypeIndex;
         }
+
+        private class ClassifiedSpanComparer : IComparer<ClassifiedSpan>
+        {
+            public static readonly ClassifiedSpanComparer Instance = new();
+
+            public int Compare(ClassifiedSpan x, ClassifiedSpan y) => x.TextSpan.CompareTo(y.TextSpan);
+        }
```

Comment:
note: as per previous discussion, i don't think this is necessary.  but if you want to keep it, that's fine.

Reply:
Interestingly enough, `GetClassifiedSpansAsync` and `AddSemanticClassificationsAsync` each return classifications in opposite order from one another. I think we might want to keep the check just in case.